DEV-INTRO.md

```diff
@@ -1,59 +1,187 @@
-PEcAn Development
-=================
+# PEcAn Development
 
-Directory Structure
--------------------
+This is a minimal guide to getting started with PEcAn development under Docker. You can find more information about docker in the [pecan documentation](https://pecanproject.github.io/pecan-documentation/master/docker-index.html).
 
-### pecan/
+## Git Repository and Workflow
 
-* modules/  Contains the modules that make up PEcAn
-* web/		The PEcAn web app to start a run and view outputs.
-* models/		Code to create the model specific configurations.
-* documentation/	Documentation about what PEcAn is and how to use it.
+We recommend following the the [gitflow](https://nvie.com/posts/a-successful-git-branching-model/) workflow and working in your own [fork of the PEcAn repsitory](https://help.github.com/en/github/getting-started-with-github/fork-a-repo). See the [PEcAn developer guide](book_source/02_demos_tutorials_workflows/05_developer_workflows/02_git/01_using-git.Rmd) for further details. In the `/scripts` folder there is a script called [syncgit.sh](scripts/syncgit.sh) that will help with synchronizing your fork with the official repository.
 
-### Modules (R packages)
+## Developing in Docker
 
-* General
-** all
-** utils
-** db
-* Analysis
-** modules/meta.analysis 
-** modules/uncertainty"
-** modules/data.land 
-** modules/data.atmosphere
-** modules/assim.batch 
-** modules/assim.sequential 
-** modules/priors
-* Model Interfaces
-** models/ed 
-** models/sipnet 
-** models/biocro
+If running on a linux system it is recommended to add your user to the docker group. This will prevent you from having to use `sudo` to start the docker containers, and makes sure that any file that is written to a mounted volume is owned by you. This can be done using `sudo adduser ${USER} docker`.
 
+To get started with development in docker we need to bring up the docker stack first. In the main pecan folder you will find the [docker-compose.yml](docker-compose.yml) file that can be used to bring up the pecan stack. There is also the [docker-compose.dev.yaml](docker-compose.dev.yaml) file that adds additional containers, and changes some services to make it easier for development.
 
-#### List of modules 
+By default docker-compose will use the files `docker-compose.yml` and `docker-compose.override.yml`. We will use the default `docker-compose.yml` file from PEcAn. The `docker-compose.override.yml` file can be used to configure it for your specific environment, in our case we will use it to setup the docker environment for development. To do this we simply rename the `docker-compose.dev.yml` file to `docker-compose.override.yml`. You can now use the command `docker-compose` to launch all the containers setup for develoopment.
```

Comment:
>we simply rename

I think it might be good to suggest using `cp` instead of `mv` otherwise they might accidentally commit deletion of the `docker-compose.dev.yml` file